File: cpp/53.最大子序和.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==1)
        {
            return nums[0];        
        }
        int max=nums[0];
        int m=0;
        int flag=0;
        for(int i=0;i<nums.size();i++)
        {
            if(flag==1)
            {
                m=0;
                flag=0;
            }
            m+=nums[i];
            if(m<0)
            {
                flag=1;
            }
            if(max<m)
            {
                max=m;
            }
        }
        return max;
    }
};

File: javascript/53.最大子序和.js
/*
 * @lc app=leetcode.cn id=53 lang=javascript
 *
 * [53] 最大子序和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
  var maxLocal = nums[0];
  var global = nums[0];
  for (let i = 1; i < nums.length; ++i) {
      maxLocal = Math.max(nums[i], nums[i] + maxLocal);
      global = Math.max(global, maxLocal);
  }
  return global;
};
// @lc code=end

